rotkehlchen/db/snapshots.py

```diff
@@ -223,41 +223,57 @@ def _export(
 
         return True, ''
 
-    def import_snapshot(
-        self,
-        balances_snapshot_file: Path,
-        location_data_snapshot_file: Path,
+    @staticmethod
+    def validate_import_data(
+        balances_data: List[Dict[str, str]],
+        location_data: List[Dict[str, str]],
+        source: str = 'csv file',
     ) -> Tuple[bool, str]:
-        """
-        Converts the snapshot files to list to dictionaries.
-        Performs a series of validation checks on the list before importing.
-        """
-        balances_list = self._csv_to_dict(balances_snapshot_file)
-        location_data_list = self._csv_to_dict(location_data_snapshot_file)
+        """Validates the snapshot data about to be imported."""
         # check if the headers match the type stored in the db
         has_invalid_headers = (
-            tuple(balances_list[0].keys()) != ('timestamp', 'category', 'asset_identifier', 'amount', 'usd_value') or  # noqa: E501
-            tuple(location_data_list[0].keys()) != ('timestamp', 'location', 'usd_value')  # noqa: E501
+            tuple(balances_data[0].keys()) != ('timestamp', 'category', 'asset_identifier', 'amount', 'usd_value') or  # noqa: E501
+            tuple(location_data[0].keys()) != ('timestamp', 'location', 'usd_value')  # noqa: E501
         )
         if has_invalid_headers:
-            return False, 'csv file has invalid headers'
+            return False, f'{source} has invalid headers'
 
         # check if all timestamps are the same.
-        balances_timestamps = [int(entry['timestamp']) for entry in balances_list]
-        location_data_timestamps = [int(entry['timestamp']) for entry in location_data_list]
+        balances_timestamps = [int(entry['timestamp']) for entry in balances_data]
+        location_data_timestamps = [int(entry['timestamp']) for entry in location_data]
         has_different_timestamps = (
             balances_timestamps.count(balances_timestamps[0]) != len(balances_timestamps) or
             location_data_timestamps.count(location_data_timestamps[0]) != len(location_data_timestamps) or  # noqa: E501
             balances_timestamps[0] != location_data_timestamps[0]
         )
         if has_different_timestamps:
-            return False, 'csv file has different timestamps'
+            return False, f'{source} has different timestamps'
 
         # check if the timestamp can be converted to int
         try:
-            _ = deserialize_timestamp(balances_list[0]['timestamp'])
+            _ = deserialize_timestamp(balances_data[0]['timestamp'])
         except DeserializationError:
-            return False, 'csv file contains invalid timestamp format'
+            return False, f'{source} contains invalid timestamp format'
+
+        return True, ''
+
+    def import_snapshot(
+        self,
+        balances_snapshot_file: Path,
```

Comment:
I basically followed the pattern used in the `accounting/export/csv.py`

Reply:
I got you. But there is a difference between them: that one is in the `accounting/export` module but this one is in the `db` module and thus is intended to work directly with database (at least in my vision). But as I see almost all other db handlers follow this pattern.

What do you think? Actually It might be okay to have it here but I would better ask Lefteris about it then.